guides/dimensiondata.md

```diff
@@ -0,0 +1,721 @@
+        DimensionDataCloudControlApi api = contextBuilder
+                .endpoint(endpoint)
+                .credentials(username, password)
+                .buildApi(DimensionDataCloudControlApi.class);
+{% endhighlight %}
+
+
+**Caution:** You will want to ensure you follow security best practices when using credentials within your code or stored in a file.
+
+## <a id="portable-abstraction-usage"></a>Portable Abstraction Usage
+
+### Terms
+Like any cloud provider, Dimension Data CloudControl has its own set of terms in cloud computing. To abstract this into jclouds' Compute interface, these terms were associated:
+
+- Node - a `Server`
+- Image - both *user-uploaded* and *provided* `CustomerImage` and `OSImage`
+- Location - `Geographic Region` equates to a jclouds Region (AU, NA, EU etc) and `Datacenter` equates to a jclouds Zone (NA9, NA12 etc). There is further information on the Geographic Regions [here](https://docs.mcp-services.net/display/CCD/Introduction+to+Geographic+Regions).
+- Hardware - number of cores, RAM size and storage size for a given Image
+
+### Getting Started
+
+{% highlight java %}
+    ComputeService compute = ContextBuilder.newBuilder( "dimensiondata-cloudcontrol" )
+                             .credentials( "username", "password" )
+                             .buildView( ComputeServiceContext.class )
+                             .getComputeService();
+{% endhighlight %}                             
+
+The implementation of the jclouds abstraction is incomplete so it is best for now to use the unwrapped *DimensionDataCloudControlApi*. 
+
+> Take note that these features are still accessible by *unwrapping* the Dimension Data CloudControl API, but this'll reduce portability of your code. See [Concepts](https://jclouds.apache.org/start/concepts/).
+
+# How To's
+## <a id="how-to-list-a-data-center"></a>How to: List a Data Center
+
+The Dimension Data CloudControl organization will be associated with one or more [MCP 2.0 Data Centers](https://docs.mcp-services.net/display/CCD/Introduction+to+MCP+2.0+Data+Center+Locations). A Data Center is a container for the assets that will be deployed and created by the jclouds Dimension Data CloudControl Provider.
+
+In jclouds terminology a Zone equates to a Data Center.
+
+The following code example shows you how to programmatically list Data Centers: 
+
+{% highlight java %}
+     PagedIterable<Datacenter> datacenters = api.getInfrastructureApi().listDatacenters();
+{% endhighlight %}
+
+This responds with a paged result set containing the Data Centers visible to your organization.
+
+## <a id="how-to-deploy-a-network-domain"></a>How to: Deploy a Network Domain
+
+For more information on Network Domains see [here](https://docs.mcp-services.net/display/CCD/Introduction+to+Cloud+Network+Domains+and+VLANs)
+
+The code to deploy a network domain follows.
+
+{% highlight java %}
+        /*
```

Comment:
Remove indent from code snippets. The code appears indented in the code box and looks ugly. Apply this to all code boxes.